packages/odyssey-react-mui/src/Form.tsx

```diff
@@ -0,0 +1,121 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { memo, ReactElement } from "react";
+import { Button, Infobox, Typography } from "./";
+
+export interface FormProps {
+  /**
+   * The title of the Form
+   */
+  title: string;
+  /**
+   * A supplementary description
+   */
+  description?: string;
+  /**
+   * The Field or FieldSet components within the Form
+   */
+  children: ReactElement | Array<ReactElement>;
+  /**
+   * An Infobox indicating a Form-wide error or status update.
+   */
+  alert?: ReactElement<typeof Infobox>;
+  /**
+   * Indicates whether input elements can by default have their values automatically completed by the browser.
+   * `autocomplete` attributes on form elements override it on <form>
+   */
+  autoComplete?: "on" | "off" | undefined;
+  /**
+   * The name of the form. The value must not be the empty string, and must be unique among the form elements in the forms collection that it is in, if any.
+   */
+  name: string;
+  /**
+   * This Boolean attribute indicates that the form shouldn't be validated when submitted.
+   * If this attribute is not set (and therefore the form is validated),
+   * it can be overridden by a formnovalidate attribute on a <button>, <input type="submit">, or <input type="image"> element belonging to the form.
+   */
+  noValidate?: boolean;
+  /**
+   * The URL that processes the form submission.
+   * This value can be overridden by a formaction attribute on a <button>, <input type="submit">, or <input type="image"> element.
+   * This attribute is ignored when method="dialog" is set.
+   */
+  action?: string;
+  /**
+   * If the value of the method attribute is post, enctype is the MIME type of the form submission.
+   * This value can be overridden by formenctype attributes on <button>, <input type="submit">, or <input type="image"> elements.
+   */
+  encType?:
+    | "application/x-www-form-urlencoded"
+    | "multipart/form-data"
+    | "text/plain"
+    | undefined;
+  /**
+   * The HTTP method to submit the form with.
+   * This value is overridden by formmethod attributes on <button>, <input type="submit">, or <input type="image"> elements.
+   */
+  method?: "post" | "get" | "dialog" | undefined;
+  /**
+   * Indicates where to display the response after submitting the form. It is a name/keyword for a browsing context (for example, tab, window, or iframe).
+   * This value can be overridden by a formtarget attribute on a <button>, <input type="submit">, or <input type="image"> element.
+   */
+  target?: "post" | "get" | "dialog" | undefined;
+  /**
+   * The Field or FieldGroup components within the Form
+   */
+  formActions: ReactElement<typeof Button> | Array<ReactElement<typeof Button>>;
```

Comment:
We did these actions differently in Dialog by passing in props because we wanted to control the render based on the requirements. We also restricted them to `Button`, but didn't specify any variants.

Are these the same styling as the ones for `Dialog`? If so, we should create an external or internal component to do both.

Reply:
The styling might match. I'll need to sync up with @taylorridgway-okta next week to confirm.

Something like `<ButtonGroup>` does seem appropriate here tho.

Reply:
Let's keep @jordankoschei-okta in the loop then.